<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Aaron Mitchell-Dick" />


<title>kallisto h5 files to tximport to DESeq2 Walkthrough</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Aaron Mitchell-Dick</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Journal.html">Journal</a>
</li>
<li>
  <a href="raw_seq_reads_to_kallsto.html">Raw Reads to kallisto</a>
</li>
<li>
  <a href="kallisto_h5_to_tximport_to_DESeq2_eval.html">Kallisto to DESeq2</a>
</li>
<li>
  <a href="About-and-References.html">About and References</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">kallisto h5 files to tximport to DESeq2 Walkthrough</h1>
<h4 class="author">Aaron Mitchell-Dick</h4>

</div>


<p><br/> <br/></p>
<div id="rna-seq-analysis-using-kallisto-tximport-and-deseq2" class="section level1">
<h1>RNA-seq analysis using kallisto, tximport, and DeSeq2</h1>
<p>This walkthrough was built by combining the information from several vignettes in order to streamline workflow from kallisto output files.<br />
To see the beginning of the workflow from raw sequencing reads (fastq files) to finished kallisto output directory, see &lt;&gt;.<br />
Here, we start with .h5 files already ordered into a directory, and we have already created a .txt files that describes the experimental paradigm, including samples and conditions.<br />
If you are executing this for the very first time, see this file &lt;&gt; for information on how to download all the packages you will need, which you’ll need to do first before you can implement the function library().</p>
<div id="load-your-libraries" class="section level3">
<h3>Load your libraries:</h3>
<pre class="r"><code>library(&quot;tximport&quot;)
library(&quot;readr&quot;)
library(&quot;tximportData&quot;)
library(&quot;rhdf5&quot;)
library(&quot;GenomicFeatures&quot;)
library(&quot;DESeq2&quot;)
library(&quot;pacman&quot;)</code></pre>
</div>
<div id="set-up-your-working-directory" class="section level3">
<h3>Set up your working directory</h3>
<p>Set the file path to the folder that contains: 1) A directory of folders wtih each of your abundance.h5 files from the kallisto operation, and contains 2) The .txt file describing samples and conditions</p>
<pre class="r"><code>setwd(&quot;C:/Users/AMDprograms/Desktop/Untrimmed/&quot;)</code></pre>
<p>Turn the folder contents into a dataframe called dir:</p>
<pre class="r"><code>dir &lt;- dir(file.path(&quot;./&quot;))
head(dir)</code></pre>
<pre><code>## [1] &quot;0h.csv&quot;              &quot;0h_STLC_vs_DMSO.csv&quot; &quot;0h_txi_kallisto.csv&quot;
## [4] &quot;5190-S1.fastq.gz&quot;    &quot;5190-S2.fastq.gz&quot;    &quot;5190-S3.fastq.gz&quot;</code></pre>
</div>
<div id="make-a-dataframe-that-describes-your-experimental-conditions" class="section level3">
<h3>Make a dataframe that describes your experimental conditions</h3>
<p>Turn the list of sample names into a dataframe:</p>
<pre class="r"><code>samples &lt;- read.table(file.path(&quot;./samples.txt&quot;), header=TRUE)
samples</code></pre>
<pre><code>##     sample condition
## 1 S1output      DMSO
## 2 S2output      STLC
## 3 S3output      DMSO
## 4 S4output      STLC
## 5 S5output      DMSO
## 6 S6output      STLC</code></pre>
<p>Make a dataframe with rows listed by sample name:</p>
<pre class="r"><code>rownames(samples) &lt;- samples$sample
samples</code></pre>
<pre><code>##            sample condition
## S1output S1output      DMSO
## S2output S2output      STLC
## S3output S3output      DMSO
## S4output S4output      STLC
## S5output S5output      DMSO
## S6output S6output      STLC</code></pre>
<p>Make a dataframe that lists the full path to all the kallisto .h5 files:</p>
<pre class="r"><code>files &lt;- file.path(&quot;./kallisto&quot;, samples$sample, &quot;abundance.h5&quot;)
files</code></pre>
<pre><code>## [1] &quot;./kallisto/S1output/abundance.h5&quot; &quot;./kallisto/S2output/abundance.h5&quot;
## [3] &quot;./kallisto/S3output/abundance.h5&quot; &quot;./kallisto/S4output/abundance.h5&quot;
## [5] &quot;./kallisto/S5output/abundance.h5&quot; &quot;./kallisto/S6output/abundance.h5&quot;</code></pre>
<p>Add column headers to the files dataframe ordered “sample[+n]”. All this does is provide columns for each sample’s reads:</p>
<pre class="r"><code>names(files) &lt;- paste0(&quot;sample&quot;, 1:6)
files</code></pre>
<pre><code>##                            sample1                            sample2 
## &quot;./kallisto/S1output/abundance.h5&quot; &quot;./kallisto/S2output/abundance.h5&quot; 
##                            sample3                            sample4 
## &quot;./kallisto/S3output/abundance.h5&quot; &quot;./kallisto/S4output/abundance.h5&quot; 
##                            sample5                            sample6 
## &quot;./kallisto/S5output/abundance.h5&quot; &quot;./kallisto/S6output/abundance.h5&quot;</code></pre>
</div>
<div id="making-the-txdb-database-to-relate-transcript-name-to-gene-name" class="section level3">
<h3>Making the TxDb database to relate transcript name to gene name</h3>
<p>Getting the right mapping for turning your kallisto list of transcripts into a list useable for tximport is important, and somewhat tricky. A few different vignettes suggested codes that did not work for me. Luckily, the authors of kallisto maintain <a href="https://github.com/pachterlab/kallisto-transcriptome-indices/releases">a database of tarball downloads</a> that not only contain high-quality pre-made kallisto indexes (.idx files) for common organisms, but also contain a .gtf file which is exactly what you can use here to make the TxDb object. So, in short, I used the kallisto tarball which includes the organism’s gtf file and to input into TxDb, making it into a TxDb dataframe. Simply navigate to the directory and file where you extracted the tarball (via Unix/Linux, on Windows Linux using WSL/Ubuntu, or on Mac) and you’re good to go.</p>
<pre class="r"><code>txdb &lt;- makeTxDbFromGFF(&quot;C:/Users/AMDprograms/Downloads/mus_musculus/mus_musculus/Mus_musculus.GRCm38.96.gtf&quot;)</code></pre>
<p>You can check what this data is like by calling these:</p>
<pre class="r"><code>columns(txdb)</code></pre>
<pre><code>##  [1] &quot;CDSCHROM&quot;   &quot;CDSEND&quot;     &quot;CDSID&quot;      &quot;CDSNAME&quot;    &quot;CDSPHASE&quot;  
##  [6] &quot;CDSSTART&quot;   &quot;CDSSTRAND&quot;  &quot;EXONCHROM&quot;  &quot;EXONEND&quot;    &quot;EXONID&quot;    
## [11] &quot;EXONNAME&quot;   &quot;EXONRANK&quot;   &quot;EXONSTART&quot;  &quot;EXONSTRAND&quot; &quot;GENEID&quot;    
## [16] &quot;TXCHROM&quot;    &quot;TXEND&quot;      &quot;TXID&quot;       &quot;TXNAME&quot;     &quot;TXSTART&quot;   
## [21] &quot;TXSTRAND&quot;   &quot;TXTYPE&quot;</code></pre>
<pre class="r"><code>keytypes(txdb)</code></pre>
<pre><code>## [1] &quot;CDSID&quot;    &quot;CDSNAME&quot;  &quot;EXONID&quot;   &quot;EXONNAME&quot; &quot;GENEID&quot;   &quot;TXID&quot;     &quot;TXNAME&quot;</code></pre>
<p>Make a key of the transcript names, make a dataframe of this key.</p>
<pre class="r"><code>k &lt;- keys(txdb, keytype = &quot;TXNAME&quot;)</code></pre>
<p>Make a dataframe using that key to map transcript names to gene names. This will allow tximport to run.</p>
<pre class="r"><code>tx_map &lt;- select(txdb, keys = k, columns=&quot;GENEID&quot;, keytype=&quot;TXNAME&quot;)</code></pre>
<pre><code>## &#39;select()&#39; returned 1:1 mapping between keys and columns</code></pre>
<pre class="r"><code>head(tx_map)</code></pre>
<pre><code>##               TXNAME             GENEID
## 1 ENSMUST00000193812 ENSMUSG00000102693
## 2 ENSMUST00000082908 ENSMUSG00000064842
## 3 ENSMUST00000192857 ENSMUSG00000102851
## 4 ENSMUST00000161581 ENSMUSG00000089699
## 5 ENSMUST00000192183 ENSMUSG00000103147
## 6 ENSMUST00000193244 ENSMUSG00000102348</code></pre>
</div>
<div id="run-tximport" class="section level3">
<h3>Run tximport</h3>
<p>Run tximport, and outputs to a dataframe txi.kallisto:</p>
<pre class="r"><code>#convert dataframe to new name
tx2gene &lt;- tx_map
# then
txi.kallisto &lt;- tximport(files, type = &quot;kallisto&quot;, tx2gene=tx2gene, ignoreTxVersion = TRUE)</code></pre>
<pre><code>## 1 2 3 4 5 6 
## transcripts missing from tx2gene: 1673
## summarizing abundance
## summarizing counts
## summarizing length
## summarizing inferential replicates</code></pre>
<pre class="r"><code>head(txi.kallisto$counts)</code></pre>
<pre><code>##                    sample1   sample2 sample3 sample4   sample5  sample6
## ENSMUSG00000000001    8299 14400.000    9423    9805 12225.000 9895.000
## ENSMUSG00000000003       0     0.000       0       0     0.000    0.000
## ENSMUSG00000000028     792  2803.000    1021    1244  1559.567 1514.813
## ENSMUSG00000000037     194   296.000     283     217   383.000  270.000
## ENSMUSG00000000049       0     0.000       0       0     0.000    0.000
## ENSMUSG00000000056    2401  3793.896    2382    2474  3247.000 2581.000</code></pre>
</div>
<div id="make-a-deseq2-object" class="section level3">
<h3>Make a DESeq2 object</h3>
<p>Turn the tximport dataset into a DeSeq2 object, called a DeSeqDataset:</p>
<pre class="r"><code>ddsTxi &lt;- DESeqDataSetFromTximport(txi.kallisto,
                                   colData = samples,
                                   design = ~ condition)
ddsTxi</code></pre>
<pre><code>## class: DESeqDataSet 
## dim: 36047 6 
## metadata(1): version
## assays(2): counts avgTxLength
## rownames(36047): ENSMUSG00000000001 ENSMUSG00000000003 ...
##   ENSMUSG00000118389 ENSMUSG00000118393
## rowData names(0):
## colnames(6): S1output S2output ... S5output S6output
## colData names(2): sample condition</code></pre>
</div>
<div id="prefilter-the-reads-to-remove-low-read-counts" class="section level3">
<h3>Prefilter the reads to remove low read counts</h3>
<p>This prefiltering step is suggested in the DESeq2 vignette. It removes any rows that have less than a total of 10 reads across all samples:</p>
<pre class="r"><code>keep &lt;- rowSums(counts(ddsTxi)) &gt;= 10
dds &lt;- ddsTxi[keep,]
dds</code></pre>
<pre><code>## class: DESeqDataSet 
## dim: 20185 6 
## metadata(1): version
## assays(2): counts avgTxLength
## rownames(20185): ENSMUSG00000000001 ENSMUSG00000000028 ...
##   ENSMUSG00000118382 ENSMUSG00000118389
## rowData names(0):
## colnames(6): S1output S2output ... S5output S6output
## colData names(2): sample condition</code></pre>
<pre class="r"><code># the above prefiltering function got rid of many transcripts (36047-x=20185)</code></pre>
</div>
<div id="make-sure-deseq2-compares-the-correct-experimental-conditions." class="section level3">
<h3>Make sure DESeq2 compares the correct experimental conditions.</h3>
<p>This next step will just make sure comparisons are made to the correct control. name the ref to whatever your control condition is named (here it’s DMSO):</p>
<pre class="r"><code>dds$condition &lt;- relevel(dds$condition, ref = &quot;DMSO&quot;)</code></pre>
</div>
<div id="run-deseq2" class="section level3">
<h3>Run DESeq2</h3>
<p>Run DeSeq2, which returns a results file “res” with log2 fold changes, p values and adjusted p values.</p>
<pre class="r"><code>dds &lt;- DESeq(dds)
res &lt;- results(dds)
head(res)</code></pre>
<pre><code>## log2 fold change (MLE): condition STLC vs DMSO 
## Wald test p-value: condition STLC vs DMSO 
## DataFrame with 6 rows and 6 columns
##                     baseMean log2FoldChange     lfcSE       stat    pvalue
##                    &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt;
## ENSMUSG00000000001 10528.932     -0.0921937  0.136806 -0.6739009  0.500374
## ENSMUSG00000000028  1402.240      0.3068577  0.290703  1.0555719  0.291164
## ENSMUSG00000000037   293.507      0.0588162  0.630455  0.0932918  0.925672
## ENSMUSG00000000056  2862.085      0.3036562  0.289039  1.0505727  0.293455
## ENSMUSG00000000058   195.899      0.7890995  0.757948  1.0410992  0.297830
## ENSMUSG00000000078  1130.841      0.3721557  0.277546  1.3408782  0.179960
##                         padj
##                    &lt;numeric&gt;
## ENSMUSG00000000001  0.999644
## ENSMUSG00000000028  0.942239
## ENSMUSG00000000037  0.999644
## ENSMUSG00000000056  0.942759
## ENSMUSG00000000058  0.944772
## ENSMUSG00000000078  0.846797</code></pre>
<p>this checks and tells you the comparison that was made (e.g. experimental vs control):</p>
<pre class="r"><code>resultsNames(dds)</code></pre>
<pre><code>## [1] &quot;Intercept&quot;              &quot;condition_STLC_vs_DMSO&quot;</code></pre>
</div>
<div id="optional-shrink-effect-size-for-making-graphs" class="section level3">
<h3>(Optional) Shrink effect size for making graphs</h3>
<p>This is going to shrink effect size, using apelgm estimator (see comment in the block for why this is used):</p>
<pre class="r"><code>resFC &lt;- lfcShrink(dds, coef=&quot;condition_STLC_vs_DMSO&quot;, type=&quot;apeglm&quot;)
head(resLFC)
# Shrinkage is not needed for text-based analysis, So the estimator
# output is written to a different variable.  This would be useful for graphing.</code></pre>
</div>
<div id="examine-the-output-of-the-differential-expression-analysis" class="section level3">
<h3>Examine the output of the differential expression analysis</h3>
<p>Order your data list in any way you’d like, here it’s ordered by pvalue:</p>
<pre class="r"><code>resOrdered &lt;- res[order(res$pvalue),]
summary(res)</code></pre>
<pre><code>## 
## out of 20185 with nonzero total read count
## adjusted p-value &lt; 0.1
## LFC &gt; 0 (up)       : 108, 0.54%
## LFC &lt; 0 (down)     : 135, 0.67%
## outliers [1]       : 387, 1.9%
## low counts [2]     : 3884, 19%
## (mean count &lt; 11)
## [1] see &#39;cooksCutoff&#39; argument of ?results
## [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p>You can filter your summary by any parameters you’d like, here its padj &lt;0.1:</p>
<pre class="r"><code>sum(res$padj &lt; 0.1, na.rm=TRUE)</code></pre>
<pre><code>## [1] 243</code></pre>
<p>Write your results to a file:</p>
<pre class="r"><code>write.csv(as.data.frame(resOrdered), 
          file=&quot;0h_STLC_vs_DMSO.csv&quot;)
#you can write this file to output but it is not as detailed as the file 
#that is output below, which contains gene descriptions.</code></pre>
<p>Check if all mapping of the data is good:</p>
<pre class="r"><code>#check if all mapping of the data is good
any(duplicated(rownames(res)))</code></pre>
<pre><code>## [1] FALSE</code></pre>
<pre class="r"><code>any(duplicated(colnames(res)))</code></pre>
<pre><code>## [1] FALSE</code></pre>
</div>
<div id="add-common-gene-names-and-short-descriptions-to-your-deseq2-output" class="section level3">
<h3>Add common gene names and short descriptions to your DESeq2 output</h3>
<p>load these libraries to prepare to add gene annotations to the dataset:</p>
<pre class="r"><code>library(&quot;AnnotationDbi&quot;)
library(&quot;org.Mm.eg.db&quot;)</code></pre>
<p>Append gene names (symbols) to the results file:</p>
<pre class="r"><code>ens.str2 &lt;- rownames(resOrdered)
head(ens.str2)</code></pre>
<pre><code>## [1] &quot;ENSMUSG00000058492&quot; &quot;ENSMUSG00000078154&quot; &quot;ENSMUSG00000056394&quot;
## [4] &quot;ENSMUSG00000027342&quot; &quot;ENSMUSG00000094777&quot; &quot;ENSMUSG00000094463&quot;</code></pre>
<pre class="r"><code>resOrdered$symbol &lt;- mapIds(org.Mm.eg.db,
                     keys=ens.str2,
                     column=&quot;SYMBOL&quot;,
                     keytype=&quot;ENSEMBL&quot;,
                     multiVals=&quot;first&quot;)</code></pre>
<pre><code>## &#39;select()&#39; returned 1:many mapping between keys and columns</code></pre>
<pre class="r"><code>head(resOrdered)</code></pre>
<pre><code>## log2 fold change (MLE): condition STLC vs DMSO 
## Wald test p-value: condition STLC vs DMSO 
## DataFrame with 6 rows and 7 columns
##                     baseMean log2FoldChange     lfcSE      stat      pvalue
##                    &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;
## ENSMUSG00000058492  192.2996       10.89298  1.361804   7.99893 1.25501e-15
## ENSMUSG00000078154  304.9936       11.55989  1.540744   7.50280 6.24704e-14
## ENSMUSG00000056394 5710.2571        1.34065  0.189609   7.07059 1.54279e-12
## ENSMUSG00000027342 8893.8889        1.14944  0.164325   6.99490 2.65442e-12
## ENSMUSG00000094777 2254.9554        1.94985  0.298221   6.53827 6.22332e-11
## ENSMUSG00000094463   97.9846        9.92236  1.596438   6.21531 5.12223e-10
##                           padj      symbol
##                      &lt;numeric&gt; &lt;character&gt;
## ENSMUSG00000058492 1.99722e-11          NA
## ENSMUSG00000078154 4.97077e-10          NA
## ENSMUSG00000056394 8.18400e-09        Lig1
## ENSMUSG00000027342 1.05606e-08        Pcna
## ENSMUSG00000094777 1.98076e-07   Hist1h2ap
## ENSMUSG00000094463 1.35859e-06          NA</code></pre>
<p>this turns the resOrdered object into a dataframe in prep for writing to csv file.</p>
<pre class="r"><code>resOrdered_df &lt;- as.data.frame(resOrdered)</code></pre>
</div>
<div id="write-your-analysis-to-a-.csv-file" class="section level3">
<h3>Write your analysis to a .csv file</h3>
<p>This now executes a block which does: 1) converts our DESeq2 results to a dataframe 2) makes a merged table that sums together gene names and descriptions</p>
<pre class="r"><code># this will make a dataframe out of merged gene names and shrot descriptions.
gene_desc &lt;- merge(org.Mm.egSYMBOL, org.Mm.egGENENAME)
# we then create a new dataframe merging the gene descriptions to the modified DeSeq2 output.
res_desc &lt;- merge(resOrdered_df, gene_desc)
head(res_desc)</code></pre>
<pre><code>##          symbol   baseMean log2FoldChange     lfcSE       stat     pvalue
## 1 0610009B22Rik  560.50309     0.86889258 0.4851068  1.7911368 0.07327135
## 2 0610010F05Rik 1084.12469    -0.18228416 0.4195712 -0.4344535 0.66395915
## 3 0610010K14Rik 3366.96582     0.17743986 0.2121151  0.8365262 0.40285892
## 4 0610012G03Rik 1737.34507    -0.05600407 0.2516677 -0.2225318 0.82389990
## 5 0610030E20Rik  673.87868    -0.19946140 0.3437019 -0.5803325 0.56169040
## 6 0610040J01Rik   36.38102    -0.57157967 2.1793335 -0.2622727 0.79311121
##        padj gene_id                  gene_name
## 1 0.6167701   66050 RIKEN cDNA 0610009B22 gene
## 2 0.9996441   71675 RIKEN cDNA 0610010F05 gene
## 3 0.9903657  104457 RIKEN cDNA 0610010K14 gene
## 4 0.9996441  106264 RIKEN cDNA 0610012G03 gene
## 5 0.9996441   68364 RIKEN cDNA 0610030E20 gene
## 6 0.9996441   76261 RIKEN cDNA 0610040J01 gene</code></pre>
<p>this writes a file that contains gene names, DiffEx data, stats, and gene descriptions all together to the current working directory.</p>
<pre class="r"><code>write.csv(res_desc, 
          file=&quot;gene_desc_0h_STLC_vs_DMSO.csv&quot;)</code></pre>
<p>Your data is now readable in R, command line, python, excel, etc.!</p>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
