<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Aaron Mitchell-Dick" />


<title>Raw Sequencing Reads to kallisto pseudoalignment</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Aaron Mitchell-Dick</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="raw_seq_reads_to_kallisto.html">Raw Reads to kallisto</a>
</li>
<li>
  <a href="kallisto_h5_to_tximport_to_DESeq2_eval.html">Kallisto to DESeq2</a>
</li>
<li>
  <a href="About-and-References.html">About and References</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Raw Sequencing Reads to kallisto pseudoalignment</h1>
<h4 class="author">Aaron Mitchell-Dick</h4>

</div>


<p><br/> <br/></p>
<div id="rna-seq-analysis-mapping-raw-sequencing-reads-to-gene-transcripts-using-kallisto-in-preparation-for-differential-expression-analysis" class="section level2">
<h2>RNA-Seq Analysis: Mapping Raw Sequencing Reads to Gene Transcripts Using kallisto in Preparation for Differential Expression Analysis</h2>
<p>This is the first part of a tutorial designed to walk you through a full RNA-Seq workflow. In this section, we will take raw sequencing reads, process them, and map them to gene transcripts. In the second section, we will take the abundance counts of gene transcripts for each sample and for each conditions, and combine them to perform differential expression analysis. To skip ahead to that section, <a href="https://aaronmitchd.github.io/RNA_Seq/kallisto_h5_to_tximport_to_DESeq2_eval.html">use this link</a>. This tutorial is designed and commented so that users with little to no experience in programming can perform robust, simple, RNA-Seq analysis. For users that desire to perform an analysis without having to write any code, see <a href="https://galaxyproject.org/">Galaxy</a>, and <a href="https://usegalaxy.org/">UseGalaxy</a>, for an open-source, web-based platform.</p>
<p><br/></p>
<div id="make-sure-you-can-run-bashlinux" class="section level3">
<h3>Make sure you can run bash/linux</h3>
<p>For Mac Users: you already run bash in the terminal so you are good to go.</p>
<p>For Windows Users: you need to download and install WSL (Windows Subsystem for Linux), then use the command $: bash on the command line to open the bash/Linux shell. The easiest way to do this is the 1) enable WSL and 2) download Ubuntu for windows by following the instructions <a href="https://ubuntu.com/wsl">here</a>. WARNING: This install can take up to an hour on a laptop. Time for a coffee break.</p>
</div>
<div id="make-sure-you-have-anacondabioconda-and-python-installed" class="section level3">
<h3>Make sure you have anaconda/bioconda and python installed</h3>
<p>An effective way to download and run bioinformatics packages is through Anaconda using bioconductor. You’ll need to <a href="https://docs.anaconda.com/anaconda/install/">install Anaconda</a>. This will install Anaconda on Mac or Windows, and will also install python. You can then run Anaconda from your programs list in Mac/Windows.</p>
</div>
<div id="install-bioconda" class="section level3">
<h3>Install bioconda</h3>
<pre class="bash"><code>conda config --add channels defaults
conda config --add channels bioconda
conda config --add channels conda-forge</code></pre>
</div>
<div id="install-all-the-programs-and-packages-youll-need-for-this-workflow" class="section level3">
<h3>Install all the programs and packages you’ll need for this workflow</h3>
<p>To see the list of programs and packages you’ll need, navigate to the <a href="">Install Resources</a> page in this tutorial. You’ll want to install all the packages, programs, and libraries you need right at the beginning, for a good flow when you’re programming. This may take a while, including installing RStudio and all the R packages for the second half of this tutorial.</p>
</div>
<div id="download-your-sequencing-files-and-run-fastqc" class="section level3">
<h3>Download your sequencing files and run FastQC</h3>
<p>Download your sequencing files (likely they are fastq.gz files) and ask your sequencing core or sequencing service what processing they performed on the reads. They likely de-multiplexed them, and may have trimmed them. If they did not do anything, you’ll need to run quality checks and trim sequencing adapters from all of your files. Sometimes they perform this step for you, and you’ll be able to tell from running FastQC what you need to do next. Download and run FastQC on each of your samples <a href="https://raw.githubusercontent.com/s-andrews/FastQC/master/INSTALL.txt">FastQC</a> and <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC website</a>.</p>
</div>
<div id="evaluate-fastqc" class="section level3">
<h3>Evaluate FastQC</h3>
<p>In the output file, you’ll find information on sequencing quality and presence or absence of adapter sequences. If any samples are of poor quality, you’ll need to make the determination whether to include them or throw them out. Then you can move forward with processing</p>
</div>
<div id="open-bash-terminal" class="section level3">
<h3>Open bash terminal</h3>
<p>On Mac open the Terminal, or on Windows open the command prompt and type ‘bash’.</p>
</div>
<div id="update-conda" class="section level3">
<h3>Update conda</h3>
<pre class="bash"><code>conda update all</code></pre>
</div>
<div id="create-a-new-conda-environment" class="section level3">
<h3>Create a new conda environment</h3>
<p>Notice that to the left of your cursor, the line begins with (base) or something like it. This is telling you what environment you are in. You want to make a new environment for each workflow you do on your computer, so that the packages you download are specific to that environment, and don’t contaminate the global environment in case versions of packages are not compatible with the version of python you have, for example. To create a new conda environment, run:</p>
<pre class="bash"><code>conda create --name RNA</code></pre>
</div>
<div id="initialize-the-conda-environment" class="section level3">
<h3>Initialize the conda environment</h3>
<p>And every time you open bash to perform this RNA-Seq analysis workflow in a new session, you’ll want to initialize this environment to run the packages you download here.</p>
<pre class="bash"><code>conda activate RNA</code></pre>
</div>
<div id="install-trimgalore" class="section level3">
<h3>Install TrimGalore</h3>
<p>TrimGalore depends on cutadapt and FastQC (optional), so you’ll need to install those. Once you are in your conda RNA environment (you can check by running ‘conda env list’ and the ’*’ will tell you which one you are currently in), install cutadapt then FastQC:</p>
<pre class="bash"><code>conda install -c bioconda cutadapt
sudo apt install fastqc</code></pre>
<p>Check versions to make sure they installed:</p>
<pre class="bash"><code>cutadapt --version 
fastqc -v</code></pre>
<p>Then install TrimGalore if you need to trim adapters and check read quality.</p>
<pre class="bash"><code>curl -fsSL https://github.com/FelixKrueger/TrimGalore/archive/0.6.5.tar.gz -o trim_galore.tar.gz
tar xvzf trim_galore.tar.gz</code></pre>
<p>To run TrimGalore with default options and run FastQC after ’[your trimgalore directory]_trim_galore –fastqc <path and filename>’:</p>
<pre class="bash"><code>/mnt/c/Users/Amdprograms/TrimGalore-0.6.5/trim_galore --fastqc &lt;filename&gt;</code></pre>
<p>If you are advanced, you can run a short for loop to run trim galore on all files in a folder, and walk away. this could take several hours depending on filesize and number.</p>
</div>
<div id="concatenate-samples-run-on-multiple-lanes" class="section level3">
<h3>Concatenate samples run on multiple lanes</h3>
<p>If you mitigated batch effects by splitting samples and running them on multiple lanes, you’ll need to concatenate those runs into a single file representing your full sample. Concatenate any samples that were run on multiple lanes using:</p>
<pre class="bash"><code>Cat 5190-S17* &gt; S17_DMSO_6h.fastq.gz
# NOTE: SAVE YOUR FILES AS fastq.gz</code></pre>
<p>Or you can even cat files from one folder and then put the output file in a different folder:</p>
<pre class="bash"><code>cat ./Aaron_data_copy/Aaron_5190_181108B1/5190-S5_S105_L00* &gt; ./Untrimmed/5190-S5.fastq.gz</code></pre>
<p>The * means ignore any other characters other than the ones displayed. This is an easy way to say “include all the files with the substring ‘<em>important_chars</em>’” to the compiler.</p>
<p>Now that you have your files, and they are pre-processed, you can run kallisto. kallisto is run in two steps:</p>
<ol style="list-style-type: decimal">
<li>generate an index file called a kallisto index file</li>
<li>pseudoalign each samples reads against the kallisto index file.</li>
</ol>
</div>
<div id="install-kallisto" class="section level3">
<h3>Install kallisto</h3>
<p>kallisto should already be installed, but if not, run</p>
<pre class="bash"><code>conda install kallisto
kallisto</code></pre>
<p>which will install kallisto, then will check version and provide a list of possible commands.</p>
</div>
<div id="create-a-kallisto-index-file" class="section level3">
<h3>Create a kallisto index file</h3>
<p>For common organisms, you can skip generating an index file, as the authors have created a repository for you. <a href="https://github.com/pachterlab/kallisto-transcriptome-indices/releases">Download the tarball for your species here</a>. It is highly recommended you follow this step if your organism is on this list, because it will make part two of this walkthrough easier. Otherwise, build your index according to the manual.</p>
</div>
<div id="extract-the-tarball" class="section level3">
<h3>Extract the tarball</h3>
<p>run this code in bash and name the destination folder.</p>
<pre class="bash"><code>sudo tar -xvzf /mnt/c/PATH/TO/TAR-FILE/Desktop/FILE-NAME.tar.gz -C /mnt/c/PATH/TO/DESTINATION/FOLDER</code></pre>
<p>Now, in the extracted folder you have your .idx file which you can just point to when running kallisto. Or, you can copy the .idx file and paste it into the folder containing all your sequencing samples. Side note, in the extracted folder you also have a .gtf file which you will use in part 2 of the walkthrough.</p>
</div>
<div id="run-kallisto" class="section level3">
<h3>Run kallisto</h3>
<p>Now, we will run kallisto on each sample. It is highly recommended you read the manual before you run your samples. Here, we will provide some simple instruction. Kallisto defaults to paired-end reads, so it will accept two files and compute the pairs. If you have single-end reads, you’ll provide an argument and it requires you to define your read length ‘-l’ and standard deviation ‘-s’ (you will know these values from your FASTQC run).</p>
<p>Navigate to the directory containing all your samples.</p>
<p>Here, we run kallisto on single-end reads with 100 bootstraps ‘-b’:</p>
<pre class="bash"><code>kallisto quant -i mouse.idx -o /mnt/c/Users/AMDprograms/Desktop/6h/sample1output -b 100 --single -l 51 -s 5 ./filename/of/working/directory.fastq.gz</code></pre>
<p>Advanced users can run a for loop to iterate through the samples. if you do this, MAKE SURE you indicate a new output folder each time. something like this ‘-o ./kallisto/sample1output’ then ‘-o ./kallist/sample2output’ otherwise you’ll just keep overwriting your abundance.h5 files.</p>
<p>Once you have folders for each sample with your kallisto output files, you are ready to move onto part two of this walkthrough!</p>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
